scripts/amoebagame.js

```diff
@@ -1,40 +1,70 @@
-            if (battles[role] && battles[role] != src && sys.isInChannel(battles[role], id)) {
-                var winner = sys.rand(0,2) > 0 ? battles[role] : src;
-                var loser = winner == src ? battles[role] : src;
-                SESSION.users(winner).amoeba.role++;
-                if (SESSION.users(loser).amoeba.role > 0) SESSION.users(loser).amoeba.role--;
-                sendChanAll("+EvolutionBot: Two " + species[role] + "s, "
-                    + sys.name(battles[role]) + " and " + sys.name(src) + " engaged in a fierce battle! "
-                    + sys.name(winner) + " won and is now a " + species[SESSION.users(winner).amoeba.role] + ". "
-                    + sys.name(loser) + " lost and is now a " + species[SESSION.users(loser).amoeba.role] + ".",
-                id);
+    this.beforeChatMessage = function (src, message, channel) {
+        if (channel !== id) {
+            return false;
+        }
+        var role = SESSION.users(src).amoeba.role,
+            isAuth = sys.auth(src) > 0 && sys.auth(src) < 4, // ignores hidden auth
+            name = htmlEscape(sys.name(src)),
+            messagePrefix = "<font color='" + script.getColor(src) + "'><timestamp/>" + pokeIcon(role) + (isAuth ? "+" : "") + "<b>" + (isAuth ? "<i>" + name + ":</i> " : name + ": ") + "</b></font>";
+        if (sys.auth(src) > 0 && ["/", "!"].indexOf(message.charAt(0)) > -1) {
+            sys.stopEvent();
+            sys.sendHtmlAll(messagePrefix + message, id);
+            return false;
+        }
+        if (role !== poke.length) { // poke.length - 1 if no battle with Arceus
+            sys.sendHtmlAll(messagePrefix + (role !== poke.length - 1 ? poke[role].cry : message), id);
+            if (battles[role] && battles[role] !== src && sys.isInChannel(battles[role], id)) {
+                var winner = sys.rand(0, 2) > 0 ? battles[role] : src,
+                    loser = winner === src ? battles[role] : src;
+                if (role < poke.length - 1) {
+                    SESSION.users(winner).amoeba.role++;
+                    if (SESSION.users(loser).amoeba.role > 0) {
+                        SESSION.users(loser).amoeba.role--;
+                    }
+                } else {
+                    SESSION.users(loser).amoeba.role = 0;
+                }
+                if (role === 0) {
+                    sys.sendHtmlAll("<hr><center>" + pokeIcon(role) + "<b>" + htmlEscape(sys.name(battles[role])) + "</b> and " + pokeIcon(role) + "<b>" + htmlEscape(sys.name(src)) + "</b> engaged in a fierce battle!</center><br /><b><font color='#0000ff'>" + htmlEscape(sys.name(winner)) + "</font></b> won and evolves into " + pokeIcon(SESSION.users(winner).amoeba.role) + poke[SESSION.users(winner).amoeba.role].species + " while <b><font color='#ff0000'>" + htmlEscape(sys.name(loser)) + "</font></b> lost and remains a " + pokeIcon(SESSION.users(loser).amoeba.role) + poke[SESSION.users(loser).amoeba.role].species + ".<hr>", id);

```

Comment:
I was trying to keep it thin while adding more color and interest with the icons. If you know how to thin it down without using much effect that be great.